### ISSUE TITLE
with_scope handled differently in wagon and engine

### ISSUE THREAD

**Issue by jaredklewis** [locomotivecms/wagon#123]
Not sure if this issues belongs here or in engine. A little background. Assume the following content types and data.

### content_types/products.yml

	---
	name: Products
	slug: products
	label_field_name: title
	fields:
		- title:
		    label: Title
		    type: string
		    required: true
		- category:
		    label: Category
		    type: belongs_to
		    class_name: categories
		    ui_enabled: true
		- note:
		    label: Note
		    type: string

### content_types/categories.yml

	---
	name: Categories
	slug: categories
	label_field_name: title
	order_by: created_at
	order_direction: asc
	fields:
		- title:
		    label: Title
		    type: string
		    required: true
		- products:
		    label: Products
		    type: has_many
		    class_name: products
		    inverse_of: category
		    ui_enabled: true

### data/products.yml

	---
	- Product 1:
	    _slug: product-1
	    category: category-1
	    note: "note a"
	- Product 2:
	    _slug: product-2
	    category: category-2
	    note: "note a"
	- Product 3:
	    _slug: product-3
	    category: category-1
	    note: "note b"

### data/categories.yml

	---
	- Category 1:
	    _slug: category-1
	- Category 2:
	    _slug: category-2

And given the following liquid markup:

	Normal:
    {% with_scope note: "note a" %}
      {% for prod in contents.products %}
      	{{ prod.title }}
      {% endfor %}
    {% endwith_scope %}
    
    Related Model:
    {% with_scope category: "category-2" %}
      {% for prod in contents.products %}
      	{{ prod.title }}
      {% endfor %}
    {% endwith_scope %}

Will result in the following ouput on Wagon:

	Normal:
        Product 1
        Product 2

    Related Model:
    	Product 2

Will result in the following ouput on Engine:

    Normal:
        Product 1
        Product 2
	
	Related Model:

So basically, using with_scope on a related model's slug  works in Wagon, but for some reason not in engine. If using with_scope in this way is not supported for technical reasons, it should definitely be documented and disabled in Wagon. Was just troubleshooting this with a real site and it took me a while to figure out what was going on.

**Comment by did** [comment]
same rendering engine (https://github.com/locomotivecms/steam) in both Engine v3 and Wagon v2.
